tests/Smoke/InstallViaComposerTest.php

```diff
@@ -164,8 +157,92 @@ public function testInstallationViaArtifactIsPossible(): void
         self::assertCommandsWork($stepsToPrepareArtifact, $tmpArtifactPath);
         self::assertCommandsWork($this->stepsToVerifyInstallation, $tmpPath);
 
-        $fs->remove($tmpPath);
-        $fs->remove($tmpArtifactPath);
+        $this->fs->remove($tmpPath);
+        $this->fs->remove($tmpArtifactPath);
+    }
+
+    /**
+     * @param array<string> $additionalCommands
+     *
+     * @dataProvider provideDoctrineAnnotationRulesetThrowsAnExceptionWhenSupportedDoctrinePackagesAreNotInstalledCases
+     */
+    public function testDoctrineAnnotationRulesetThrowsAnExceptionWhenSupportedDoctrinePackagesAreNotInstalled(array $additionalCommands): void
+    {
+        $tmpPath = $this->createFakeComposerProject($this->currentCodeAsComposerDependency);
+
+        self::assertCommandsWork(
+            [
+                'composer install -q',
```

Comment:
I this really `install` or impilcit `update`?

If the latter then if we move `...$additionalCommands,` above it we can add `--no-update` to `composer require` in the data provider to avoid installing/solving dependencies twice.

Reply:
We need to ask Mr. Copy Paste 😅.